### PR TITLE
Start automatically generating trace data structure

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/656

Changes in this PR:
* Converted SpanStatus and SpanEvent.
* Start using AttributeMap in trace (everywhere)
* Removed Attributes

### CODE REVIEW

File: internal/data_generator/internal/trace_structs.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package internal
+
+var traceFile = &File{
+	Name: "trace",
+	imports: []string{
+		`otlptrace "github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1"`,
+	},
+	structs: []baseStruct{
+		spanEvent,
+		//SpanLink,
```
Comment (pjanotti) [inline internal/data_generator/internal/trace_structs.go]: nit: delete  commented code

File: internal/data_generator/internal/trace_structs.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		spanStatus,
+	},
+}
+
+var spanEvent = &messageStruct{
+	structName: "SpanEvent",
+	description: "// SpanEvent is a time-stamped annotation of the span, consisting of user-supplied\n" +
```
Comment (pjanotti) [inline internal/data_generator/internal/trace_structs.go]: nit: it seems that the generator already adds "// SpanEvent" see stutter on generated code. Same for the next struct.

Comment (bogdandrutu) [reply internal/data_generator/internal/trace_structs.go]: I fixed that in #670. Will get regenerated this file.

File: internal/data_generator/internal/trace_structs.go (hunk 1 above)
Comment (bogdandrutu) [reply internal/data_generator/internal/trace_structs.go]: Done.

**Pull request merged**